### PR TITLE
Sttaging bug fixes

### PR DESCRIPTION
**Pull request by Bod9001**
### Changelog:

CL: [New] Adds a Italian Speech modifier to fake moustache
CL: [Improvement] improves the logic for graffiti and Mopping graffiti off walls

CL: [Fix] Fixes door not opening on bump interaction
CL: [Fix] Fixes graffiti rendering above the player
CL: [Fix] Fixes speech modifier bug not applying on the last word of a chat message

### CODE REVIEW

File: UnityProject/Assets/Scripts/HealthV2/Living/Damage/Trauma/TraumaTypes/TraumaBurning.cs [hunk 1]
```diff
@@ -67,7 +67,7 @@ public override void ProgressDeadlyEffect()
 					{
 						currentStage = 3;
 						Chat.AddActionMsgToChat(bodyPart.HealthMaster.gameObject,
-							$"<color=red>Parts of the {bodyPart.gameObject.ExpensiveName()} crumble and ash away.</color");
+							$"<color=red>Parts of the {bodyPart.gameObject.ExpensiveName()} crumble and ash away.</color>");
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/HealthV2/Living/Damage/Trauma/TraumaTypes/TraumaBurning.cs]: REEEEEEEEEEEEEEEEEE use the rich text thingy I created for exactly this

File: UnityProject/Assets/Scripts/Items/Tool/Mop.cs [hunk 2]
```diff
@@ -41,12 +41,6 @@ public bool WillInteract(PositionalHandApply interaction, NetworkSide side)
 		//can only mop tiles
 		if (!Validations.HasComponent<InteractableTiles>(interaction.TargetObject)) return false;
 
-		//don't attempt to mop walls
-		if (MatrixManager.IsWallAt(interaction.WorldPositionTarget.RoundToInt(), isServer: side == NetworkSide.Server))
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Items/Tool/Mop.cs:45]: why is this removed?

File: UnityProject/Assets/Scripts/ScriptableObjects/SpeechModifier.cs [hunk 3]
```diff
@@ -57,9 +57,20 @@ string Replace(string message)
...
 				char c = message[i];
 
-				if (!char.IsSeparator(c) && !char.IsPunctuation(c)) continue;
 
-				string substring = message.Substring(start, i - start);
+				if (!char.IsSeparator(c) && !char.IsPunctuation(c) && (i != message.Length -1) ) continue;
+
+				string substring = "";
+
+				if ((i == message.Length - 1))
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/ScriptableObjects/SpeechModifier.cs]: why the double (())

File: UnityProject/Assets/Scripts/ScriptableObjects/SpeechModifier.cs [hunk 4]
```diff
@@ -57,9 +57,20 @@ string Replace(string message)
 			{
 				char c = message[i];
 
-				if (!char.IsSeparator(c) && !char.IsPunctuation(c)) continue;
 
-				string substring = message.Substring(start, i - start);
+				if (!char.IsSeparator(c) && !char.IsPunctuation(c) && (i != message.Length -1) ) continue;
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/ScriptableObjects/SpeechModifier.cs]: ! is for inverting in this case only reee

**Review by MaxIsJoe — APPROVED** [review]

File: UnityProject/Assets/Scripts/HealthV2/Living/Damage/Trauma/TraumaTypes/TraumaBurning.cs (hunk 1 above)
Comment (corp-0) [inline UnityProject/Assets/Scripts/HealthV2/Living/Damage/Trauma/TraumaTypes/TraumaBurning.cs]:
```suggestion
$"Parts of the {bodyPart.gameObject.ExpensiveName()} crumble and ash away".Color(Color.Red);
```

**Comment by MaxIsJoe** [comment]
when did i approve this
no

**Review by MaxIsJoe — CHANGES_REQUESTED** [review]
reviewws

File: UnityProject/Assets/Scripts/Items/Tool/Mop.cs (hunk 2 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Tool/Mop.cs:45]: So you can clean graffiti on walls

**Review by MaxIsJoe — APPROVED** [review]

**Pull request merged**